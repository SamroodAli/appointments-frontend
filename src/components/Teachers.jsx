import { useNavigate } from 'react-router-dom';
import getTeachers from '../api/getTeachers';
import HorizontalScroll from './horizontal-scroll';
import useData from '../hooks/useData';

const Teachers = () => {
  const navigate = useNavigate();

  const { notReady, notReadyContent, data } = useData('teachers', getTeachers);

  if (notReady) {
    return notReadyContent;
  }

  // id has to be String for horizontal-scroll to work
  // shuffle the array
  const items = data
    .sort(() => Math.random() - 0.5)
    .map((teacher, idx) => ({
      id: teacher.id,
      name: teacher.name,
      index: idx,
    }));

  const onItemClick = (item) => {
    navigate(`/teachers/${item.id}/${item.index}`);
  };

  return (
    <div className="text-center grid grid-cols-1 items-center h-full">
      <h1 className="font-serif text-5xl lg:text-5xl mb-4">Best in class Teachers</h1>
      <HorizontalScroll items={items} onItemClick={onItemClick} />
    </div>
  );
};
export default Teachers;
