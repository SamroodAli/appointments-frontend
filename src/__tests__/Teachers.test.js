import renderer from 'react-test-renderer';
// import { render } from '@testing-library/react';
import Teachers from '../components/Teachers';
import Wrapper from '../testWrapper';

test('Teachers snapshot testing', () => {
  const tree = renderer.create(
    <Wrapper>
      <Teachers />
    </Wrapper>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
