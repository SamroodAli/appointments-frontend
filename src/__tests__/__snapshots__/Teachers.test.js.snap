// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teachers snapshot testing 1`] = `
<button
  className="bg-yellow-500 p-2  top-0 w-full"
  disabled={true}
  type="button"
>
  <p
    className=" font-mono animate-pulse"
  >
    Loading
  </p>
</button>
`;
